src/render.rs

```diff
@@ -471,31 +471,40 @@ pub fn draw_curves(
         gaplength,
         minimumgap,
         label_depressions,
+        remove_touching_contours,
         ..
     } = config;
     formlinesteepness *= scalefactor;
 
     let mut size: f64 = 0.0;
     let mut xstart: f64 = 0.0;
     let mut ystart: f64 = 0.0;
-    let mut x0: f64 = 0.0;
-    let mut y0: f64 = 0.0;
+    let tfw_in = tmpfolder.join("vegetation.pgw");
+    let mut lines = BufReader::new(fs.open(tfw_in).expect("PGW file does not exist")).lines();
+    let x0 = lines
+        .nth(4)
+        .expect("no 4 line")
+        .expect("Could not read line 5")
+        .parse::<f64>()
+        .unwrap();
+    let y0 = lines
+        .next()
+        .expect("no 5 line")
+        .expect("Could not read line 6")
+        .parse::<f64>()
+        .unwrap();
     let mut steepness: HashMap<(usize, usize), f64> = HashMap::default();
 
-    if formline > 0.0 {
-        let heightmap_in = tmpfolder.join("xyz2.hmap");
-        let mut reader = BufReader::new(fs.open(heightmap_in)?);
-        let hmap = HeightMap::from_bytes(&mut reader)?;
+    let heightmap_in = tmpfolder.join("xyz2.hmap");
+    let mut reader = BufReader::new(fs.open(heightmap_in)?);
+    let hmap = HeightMap::from_bytes(&mut reader)?;
+    let xyz = &hmap.grid;
```

Comment:
Instead of reading the PGW file, perhaps we should take the `x0` and `y0` directly from the loaded `hmap`:
```rust
let x0 = hmap.xoffset;
let y0 = hmap.yoffset;
```

Reply:
Also, one renrering trick of old was editing vegetation png image (canvas size) and pgw file values and re-renderding everything to get the exact tile extent we wanted (like 3x3 km) even if laz tile was smaller for being near sea shore or area edge. Using png and pgw might enable that possibility again.

Reply:
@rphlo sheet 592525 
https://geoportaal.maaamet.ee/index.php?lang_id=2&plugin_act=otsing&kaardiruut=592525&andmetyyp=lidar_laz_tava&dl=1&f=592525_2024_tava.laz&no_cache=676c5ef71b4d3&page_id=664

Reply:
![kuva](https://github.com/user-attachments/assets/b117fbc9-bd2e-4052-9729-e0a8c47a0641)

Reply:
I downloaded that laz tile and also neighbring tiles and that offset error does not happen... But i can remember seeing this issue and pgw thing fixed it then. I wonder why it is not happening now, less laz files in input folder for sure but still. It may be better not change anything and take a look at this after someone facing this issue again.

Reply:
Now I could reproduce it. This laz file:
https://geoportaal.maaamet.ee/index.php?lang_id=2&plugin_act=otsing&kaardiruut=592525&andmetyyp=lidar_laz_tava&dl=1&f=592525_2020_tava.laz&no_cache=676c768990f02&page_id=664

Reply:
Alright that does make sense. I guess we should use the min/max dimensions of the input Laz file instead of the area the ground points cover when generating the heightmap... A bit like the fix @rphlo just added. But I guess for batch mode where we extend the edges with points from neighboring tiles that will not really work... Perhaps hard-coding the dimension offset would work in that case (just like we do already with `+/-127m`. For now using the PGW file is okay and we can work on making the handling of empty areas better during heightmap generation as a next step :rocket:

Reply:
I think we might need ini settings for tile width and height (meters) and tileset origo xy coordinates. And renderer would calculate the tile extent it is processing and render the full tile even if only part of it has laz data. Norway and Estonia has 1x1 km tiles, Finland 3x3 km tiles Sweden 2.5x2.5km tiles I think. The old exe used for those projects had some hard coded tweaks for that, but it should be configurable really.

For estonia I had to have a post rendering step for smacking maa-amet basic map rasters and KP result together and that took care of that issue.